ui/src/app/routing/guards.tsx

```diff
@@ -28,26 +31,54 @@ const userIsEnabled = (userDisabledInDb: boolean) =>
   !userDisabledInDb &&
   eligibleForTier(profileStore.get().profile, AccessTierShortNames.Registered);
 
+// TODO use redirectTo?
 export const disabledGuard = (userDisabledInDb: boolean): Guard => ({
   // Show disabled screen when user account is disabled by admin or removed from institution registered tier requirement.
   allowed: (): boolean => userIsEnabled(userDisabledInDb),
   redirectPath: '/user-disabled',
 });
 
+// TODO use redirectTo?
 export const userDisabledPageGuard = (userDisabledInDb: boolean): Guard => ({
   // enabled users should be redirected to the homepage if they visit the /user-disabled page
   allowed: (): boolean => !userIsEnabled(userDisabledInDb),
   redirectPath: '/',
 });
 
+// where a user should be redirected to, depending on the current status of their access modules
+export enum AccessModulesRedirection {
+  NO_REDIRECT,
+  DATA_ACCESS_REQUIREMENTS,
+  ACCESS_RENEWAL,
+}
+
+// use this for all access-module routing decisions, to ensure only one routing is chosen
+export const redirectTo = (profile: Profile): AccessModulesRedirection => {
```

Comment:
This initial implementation matches the existing behavior: expiration is checked first (using the same expiration check) before the general RT membership check (which also uses the same membership check).